examples/_HeatDiffusion/ExampleADHeatDiffusion.jl

```diff
@@ -0,0 +1,359 @@
+# The other does it with Enzyme: https://github.com/EnzymeAD/Enzyme.jl
+
+# Computation of the partial derivatives of the residual function. Sets the coefficient in the system matrix.
+function AssemblyPoisson_Enzyme!(K, u, k, s, number, type, pattern, bc_val, nc, Δ, u0, ρ, cp)
+
+    # This is aocal matrix that stores the partial derivatives.
+     
+    # Remember:     #             0    uN    0
+                    #     u_loc = uW   uC    uE
+                    #             0    uS    0 
+                    
+    # Now we have   #                0     df_duN      0 
+                    #     ∂R∂u  = df_duW   df_duC   df_duE
+                    #                0     df_duS      0 
+    # These cofficient are automatically computed by Enzyme and are then collected in the sparse matrix that defines the linear system of equations
+    ∂R∂u     = @MMatrix zeros(3,3) 
+    shift    = (x=1, y=1)
+
+    # Same as above 
+    k_loc_shear = @SVector(zeros(2))
+
+    # Loop over the cells in 2D
+    for j in 1+shift.y:nc.y+shift.y, i in 1+shift.x:nc.x+shift.x
+
+        # This is a 3*3 matrix that contains the number of each equation in the stencil (each cell has one corresponding value of u, which has a unique equation number)
+        num_loc   = SMatrix{3,3}(number.u[ii,jj] for ii in i-1:i+1, jj in j-1:j+1) .* pattern.u.u
+
+        # This is the 3*3 matrix that contains the current stencil values (see residual function above) 
+        u_loc     = MMatrix{3,3}(u[ii,jj] for ii in i-1:i+1, jj in j-1:j+1)
+
+        # Same as above: conductivity tensor
+        k_loc_xx  = @SVector [k.x.xx[i-1,j-1], k.x.xx[i,j-1]]
+        k_loc_yy  = @SVector [k.y.yy[i-1,j-1], k.y.yy[i-1,j]]
+        k_loc     = (xx = k_loc_xx,    xy = k_loc_shear,
+                     yx = k_loc_shear, yy = k_loc_yy)
+
+        # Same as above: boundary condition VALUES
+        bcv_loc   = SMatrix{3,3}(bc_val.u[ii,jj] for ii in i-1:i+1, jj in j-1:j+1)
+        # Same as above: boundary condition TYPES
+        type_loc  = SMatrix{3,3}(  type.u[ii,jj] for ii in i-1:i+1, jj in j-1:j+1)
+
+        # Initialise partial derivatives to 0
+        ∂R∂u     .= 0e0
+
+        # Here the magic happens: we call a function from Enzyme that computes all, partial derivatives for the current stencil block 
+        autodiff(Enzyme.Reverse, Diffusion2D, Duplicated(u_loc, ∂R∂u), Const(k_loc), Const(s[i,j]), Const(type_loc), Const(bcv_loc), Const(Δ), Const(u0[i,j]), Const(ρ[i,j]), Const(cp[i,j]))
+
+        # This loops through the 2*2 stencil block and sets the coefficient ∂R∂u into the sparse matrix K.u.u
+        num_ij = number.u[i,j]
+        for jj in axes(num_loc,2), ii in axes(num_loc,1)
+            if num_loc[ii,jj] > 0 # only if equation exists, then put a value
+                # For the given equation number (num_ij) enter all values on the line of the sparse matrix K.u.u
+                K.u.u[num_ij, num_loc[ii,jj]] = ∂R∂u[ii,jj]  
+            end
+        end
```

Comment:
```suggestion
        for I in eachindex(num_loc)
            if num_loc[I] > 0 # only if equation exists, then put a value
                # For the given equation number (num_ij) enter all values on the line of the sparse matrix K.u.u
                K.u.u[num_ij, num_loc[I]] = ∂R∂u[I]
            end
        end
```
Could simplify it a bit like this.